validator/sawtooth_validator/journal/block_validator.py

```diff
@@ -185,73 +180,65 @@ def _get_previous_block_state_root(self, blkw):
 
         return self._block_cache[blkw.previous_block_id].state_root_hash
 
-    @staticmethod
-    def _validate_transactions_in_batch(batch, chain_commit_state):
-        """Verify that all transactions in this batch are unique and that all
-        transaction dependencies in this batch have been satisfied.
-
-        :param batch: the batch to verify
-        :param chain_commit_state: the current chain commit state to verify the
-            batch against
-        :return:
-        Boolean: True if all dependencies are present and all transactions
-        are unique.
+    def _validate_batches_in_block(self, blkw, prev_state_root):
         """
-        for txn in batch.transactions:
-            txn_hdr = TransactionHeader()
-            txn_hdr.ParseFromString(txn.header)
-            if chain_commit_state.has_transaction(txn.header_signature):
-                LOGGER.debug(
-                    "Batch invalid due to duplicate transaction: %s",
-                    txn.header_signature[:8])
-                return False
-            for dep in txn_hdr.dependencies:
-                if not chain_commit_state.has_transaction(dep):
-                    LOGGER.debug(
-                        "Batch invalid due to missing transaction dependency;"
-                        " transaction %s depends on %s",
-                        txn.header_signature[:8], dep[:8])
-                    return False
-        return True
+        Validate all batches in the block. This includes:
+            - Validating all transaction dependencies are met
+            - Validating there are no duplicate batches or transactions
+            - Validating execution of all batches in the block produces the
+              correct state root hash
 
-    def _validate_batches_in_block(
-        self, blkw, prev_state_root, chain_commit_state
-    ):
+        Args:
+            blkw: the block of batches to validate
+            prev_state_root: the state root to execute transactions on top of
+
+        Raises:
+            BlockValidationError:
+                If validation fails, raises this error with the reason.
+            MissingDependency:
+                Validation failed because of a missing dependency.
+            DuplicateTransaction:
+                Validation failed because of a duplicate transaction.
+            DuplicateBatch:
+                Validation failed because of a duplicate batch.
+        """
         if blkw.block.batches:
```

Comment:
In order to save a level of indentation, it would be nice if this whole section were rewritten as

```
if not blkw.block.batches:
    return

try:
    blah blah blah
```